src/ProxyClient/Varnish.php

```diff
@@ -133,26 +142,28 @@ public function refresh($url, array $headers = array())
      *                              refresh and no base URL is set
      *
      */
-    protected function createRequest($method, $url, array $headers = array())
+    protected function queueRequest($method, $url, array $headers = [])
     {
-        $request = parent::createRequest($method, $url, $headers);
+        $request = new InvalidationRequest($method, $url, $headers);
 
-        // For purge and refresh, add a host header to the request if it hasn't
-        // been set
         if (self::HTTP_METHOD_BAN !== $method
-            && '' == $request->getHeader('Host')
+            && !$this->baseUriSet
+            && !$request->getHeaderLine('Host')
         ) {
             throw MissingHostException::missingHost($url);
         }
 
-        return $request;
+        parent::queueRequest($method, $url, $headers);
     }
 
     /**
      * {@inheritdoc}
+     *
+     * Additionally, store whether a base URI was set
      */
-    protected function getAllowedSchemes()
+    protected function initQueue(array $servers, $baseUri)
     {
-        return array('http');
+        $this->queue = new RequestQueue($servers, $baseUri);
+        $this->baseUriSet = $baseUri !== null;
```

Comment:
could we instead check that information on the queue?

and why is that only relevant for varnish but not for the other clients? don't all clients need to know to what server they should talk?